services/src/floodlight-modules/src/main/java/org/openkilda/floodlight/switchmanager/SwitchManager.java

```diff
@@ -778,6 +780,29 @@ public void installDropFlow(final DatapathId dpid) throws SwitchOperationExcepti
         }
     }
 
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public void installDropLoopRule(final DatapathId dpid) throws SwitchOperationException {
```

Comment:
it is much more efficient to get IOFSwitch object in argument instead of DatapathId... all methods called from installDefaultRules call lookupSwitch at start... this is too inefficiently.

Reply:
Agree, but let's do it in a separate PR